python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -2351,3 +2351,22 @@ def test_array_from_pylist_offset_overflow():
     assert isinstance(arr, pa.ChunkedArray)
     assert len(arr) == 2**31
     assert len(arr.chunks) > 1
+
+
+def check_run_end_encoded_with_type(ree_type=None):
```

Comment:
Can you move this to `test_array.py` ? (I _think_ that's a better fit for testing the attributes of the array class and the from_arrays constructor)

Reply:
And could you also add a test for `RunEndEncodedArray.from_buffers`? (this gets inherited from the base class, but we should ensure this version actually works and not segfaults or so)

Reply:
There is a children keyword, so I assume it would take an empty list for the buffers, and a list of children

Reply:
I wonder if I should remove the `=None` from `children=None` in my override version of `from_buffers`. What do you say?